API_doc/Overview.md

```diff
@@ -0,0 +1,59 @@
+# Creating and retrieving retrieving a configuration
+
+The first call $LIB_NAME should be to the following function:
+```cpp
+$LIB_NAME::Config $LIB_NAME::createConfig(char const *configName);
+```
+`configName` the name given to the configuration.
+
+```cpp
+struct $LIB_NAME::Id
+{
+  void *data;
+  size_t size; // size of data
+};
+
+$LIB_NAME::Id $LIB_NAME::Config::getId();
+$LIB_NAME::Config $LIB_NAME::getConfig($LIB_NAME::Id id);
+```
+
+Using the above functions it's possible for an application to persistently use one configuration once it has been created.
+
+# Modifying a configuration
+
+A configuration is represented by a series of instructions.
+The most basic instruction is a setting, which can alias another setting.
+
+```cpp
+void $LIB_NAME::Config:addSetting(char const *name, char const *value);
+void $LIB_NAME::Config:addSettingAlias(char const *name, char const *aliasName);
```

Comment:
Est ce qu'on ne retournerait pas une `enum` pour ces deux fonctions afin d'avertir l'utilisateur d'un problème ? (`name` existe déjà, il y a eu un problème lors de l'accès au fichier de sauvergarde, tout s'est bien passé, etc)

Reply:
I forgot to write:
In the case the setting already exists, it is overridden.
About access / connection errors etc. I think they could be unified across the API ? This is something we have to think about.